packages/auth/src/client/service.ts

```diff
@@ -166,14 +276,126 @@ function verifyClientDisplay(
   )
 }
 
-function verifyJwk(jwk: JWKWithRequired, keys: RegistryKey): boolean {
-  // TODO: update this to reflect eventual shape of response from registry
+function isJwkViable(keys: RegistryKey): boolean {
   return !!(
-    !keys.revoked &&
-    keys.exp &&
-    new Date() < new Date(keys.exp * 1000) &&
-    keys.nbf &&
-    new Date() >= new Date(keys.nbf * 1000) &&
-    keys.x === jwk.x
+    (!keys.exp || new Date() < new Date(keys.exp * 1000)) &&
+    (!keys.nbf || new Date() >= new Date(keys.nbf * 1000))
   )
 }
+
+function verifyJwk(jwk: JWKWithRequired, keys: RegistryKey): boolean {
+  // TODO: update this to reflect eventual shape of response from registry
+  return !!(!keys.revoked && isJwkViable(keys) && keys.x === jwk.x)
+}
+
+async function tokenHttpsigMiddleware(
+  deps: ServiceDependencies,
+  ctx: AppContext,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  next: () => Promise<any>
+): Promise<void> {
+  const sig = ctx.headers['signature']
+  const sigInput = ctx.headers['signature-input']
+
+  if (
+    !sig ||
+    !sigInput ||
+    typeof sig !== 'string' ||
+    typeof sigInput !== 'string'
```

Comment:
I'm wondering if the openapi schema can enforce this?

Reply:
I tried looking to see if it can validate request headers but I wasn't able to find a way to describe that in the spec from the OpenAPI docs.